scipp/qens-reduction.ipynb

```diff
@@ -85,46 +192,135 @@
   {
    "cell_type": "markdown",
    "id": "c3dd4d68-692c-4eb9-a92d-56907b362ed7",
-   "metadata": {},
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
    "source": [
-    "## Mask bad region"
+    "### Exercise 1: mask bad region\n",
+    "\n",
+    "We need to handle the events that are affected by the detector defect.\n",
+    "The simplest way of doing so it to mask the affected region in y.\n",
+    "We will do so by binning in y and applying a y-dependent mask to the data array.\n",
+    "\n",
+    "To this end, define bin-edges in the y-dimension that cover the entire range of the data.\n",
+    "There should be three bins which cover the y-range below the broken region, the broken region, and the the range above the broken region, respectively.\n",
+    "So you should end up with a length-4 array of bin-edges.\n",
+    "\n",
+    "Then, define another array for the mask which should have the values `[False, True, False]`.\n",
+    "So it masks the middle bin but not the others.\n",
+    "\n",
+    "Finally, apply the mask by binning the events using the edges you defined and inserting the mask into the binned array's masks.\n",
+    "\n",
+    "**Hints**:\n",
+    "\n",
+    "- Use `events.coords[\"y\"].min().value` and `np.nextafter(events.coords[\"y\"].max().value, np.inf)` to get the bounds in y. (The call to `np.nextafter` is needed because we need the position *after* the final event for the bin-edges.)\n",
```

Comment:
We could. But since there is a significant density of events all the way to the tof boundaries, extending the range would lead to a sharp drop off. From what I understand, this wouldn't affect the analysis. But it wouldn't be the best solution.

On the other hand, sacrificing a few events at the edge wouldn't ultimately matter. So we could just have them guess a conservative edge.

Reply:
> On the other hand, sacrificing a few events at the edge wouldn't ultimately matter. So we could just have them guess a conservative edge.

yes, i think going for simplicity is best.